pkg/sdkserver/sdkserver.go

```diff
@@ -1243,27 +1243,35 @@ func (s *SDKServer) sendGameServerUpdate(gs *agonesv1.GameServer) {
 	s.streamMutex.Lock()
 	defer s.streamMutex.Unlock()
 
-	for i, stream := range s.connectedStreams {
+	// Filter the slice of streams sharing the same backing array and capacity as the original
+	// so that storage is reused and no memory allocations are made. This modifies the original
+	// slice.
+	//
+	// See https://go.dev/wiki/SliceTricks#filtering-without-allocating
+	remainingStreams := s.connectedStreams[:0]
```

Comment:
Ok, I get what you're doing, but do we really care? If you have to document it this thoroughly, are we sure it's worth the optimization?

Reply:
I believe that this loop gets executed quite frequently (every time we send a game server change event, so when game servers move to ready, allocated, shutdown, etc or have labels set, counts or lists modified, etc.). 

Code-wise, it isn't much different from allocating a new slice, appending to it, and then replacing the `connectedStreams` (the only different being line 1251 which would change to `remainingStreams := make(...)`. So making it more memory efficient doesn't seem to have any downsides other than this extra long comment.